build/utils.py

```diff
@@ -14,133 +14,34 @@
-            job_config_dict = json.load(f)
-    elif json_env_var:
-        job_config_dict = txt_to_obj(json_env_var)
-    else:
-        raise ValueError(
-            "Must set environment variable 'SFT_TRAINER_CONFIG_JSON_PATH' \
-        or 'SFT_TRAINER_CONFIG_JSON_ENV_VAR'."
-        )
-    return job_config_dict
 
+def get_highest_checkpoint(dir_path):
+    checkpoint_dir = ""
+    for curr_dir in os.listdir(dir_path):
+        if curr_dir.startswith("checkpoint"):
+            if checkpoint_dir:
+                curr_dir_num = int(checkpoint_dir.rsplit("-", maxsplit=1)[-1])
+                new_dir_num = int(curr_dir.split("-")[-1])
+                if new_dir_num > curr_dir_num:
+                    checkpoint_dir = curr_dir
+            else:
+                checkpoint_dir = curr_dir
 
-def process_launch_training_args(job_config_dict):
-    """Return parsed config for tuning to pass to SFT Trainer
-    Args:
-        job_config_dict: dict
-    Return:
-        model_args: configs.ModelArguments
-        data_args: configs.DataArguments
-        training_args: configs.TrainingArguments
-        tune_config: peft_config.LoraConfig | peft_config.PromptTuningConfig
-        merge_model: bool
-        file_logger_config: tracker_configs.FileLoggingTrackerConfig
-        aim_config: tracker_configs.AimConfig
-    """
-    parser = transformers.HfArgumentParser(
-        dataclass_types=(
-            configs.ModelArguments,
-            configs.DataArguments,
-            configs.TrainingArguments,
-            peft_config.LoraConfig,
-            peft_config.PromptTuningConfig,
-            tracker_configs.FileLoggingTrackerConfig,
-            tracker_configs.AimConfig,
-        )
-    )
-
-    (
-        model_args,
-        data_args,
-        training_args,
-        lora_config,
-        prompt_tuning_config,
-        file_logger_config,
-        aim_config,
-    ) = parser.parse_dict(job_config_dict, allow_extra_keys=True)
-
-    peft_method_parsed = job_config_dict.get("peft_method")
-
-    tune_config = None
-    merge_model = False
-    if peft_method_parsed == "lora":
-        tune_config = lora_config
-        merge_model = True
-    elif peft_method_parsed == "pt":
-        tune_config = prompt_tuning_config
-
-    logging.info(
-        "Parameters used to launch training: \
-    model_args %s, data_args %s, training_args %s, tune_config %s \
-        file_logger_config %s aim_config %s",
-        model_args,
-        data_args,
-        training_args,
-        tune_config,
-        file_logger_config,
-        aim_config,
-    )
-
-    return (
-        model_args,
-        data_args,
-        training_args,
-        tune_config,
-        merge_model,
-        file_logger_config,
-        aim_config,
-    )
+    return checkpoint_dir
+
+
+def serialize_args(args_json):
+    message_bytes = pickle.dumps(args_json)
+    base64_bytes = base64.b64encode(message_bytes)
+    return base64_bytes.decode("ascii")
```

Comment:
I separated them because I'm trying to separate what is run within the image (scripts in `build/`) versus what is run when the library is directly called (in `tuning/`). You are absolutely right that they are doing the same but opposite functions but  `serialize_args()` is only called in `accelerate_launch.py` whereas `txt_to_obj()` is called in `get_json_config()` which is called both in `accelerate_launch.py` and `sft_trainer.py`